pkg/cassandra/datacenter.go

```diff
@@ -348,9 +348,61 @@ func Coalesce(clusterName string, clusterTemplate *api.CassandraClusterTemplate,
 	dcConfig.ClientEncryptionStores = clusterTemplate.ClientEncryptionStores
 	dcConfig.AdditionalSeeds = clusterTemplate.AdditionalSeeds
 
+	if len(dcTemplate.DatacenterOptions.AdditionalContainers) > 0 {
```

Comment:
The `Coalesce` function previously had a single purpose - combine the cluster and DC templates with override semantics. Now the function is also modifying the PodTemplateSpec and AdditionalVolumes. It would be nice to keep the function single-purposed and do the modifications downstream.

Reply:
The problem is that the `Coalesce` function is both:
- where the dcConfig object is initialized
- where we deal with overlaps between cluster and dc level settings

Since we need to do the latter, we end up doing the former as well.
I guess what would be right to do here is rather add `AdditionalInitContainers` and `AdditionalContainers` to the `DatacenterConfig` struct and fill them in the `Coalesce` function but only modify the `PodTemplateSpec` afterwards outside of it.
Sounds good?

Reply:
Sounds good